superset-frontend/src/dashboard/util/charts/getEffectiveExtraFilters.ts

```diff
@@ -21,7 +21,7 @@ import { DataRecordFilters } from '@superset-ui/chart';
 export default function getEffectiveExtraFilters(filters: DataRecordFilters) {
   return Object.entries(filters).map(([column, values]) => ({
     col: column,
-    op: 'in',
+    op: Array.isArray(values) ? 'IN' : '=',
```

Comment:
The current logic generates an invalid filter (picked up by TS linter in `superset-ui`).